docs/react-v9/contributing/rfcs/shared/build-system/stop-styles-transforms.md

```diff
@@ -0,0 +1,162 @@
+# RFC: Stop pre-processing styles with Griffel in `@fluentui/react-components`
+
+[@layershifter](https://github.com/layershifter)
+
+## Background
+
+We've begun noticing a pattern where apps are bundled into multiple separate bundles. For example, there's the main bundle for the app itself and a separate bundle for a first UI piece fetched from a CDN. This problem is described in the following issues:
+
+- [microsoft/griffel#453](https://github.com/microsoft/griffel/issues/453) (_check this issue for more details_)
+- [microsoft/griffel#526](https://github.com/microsoft/griffel/issues/526)
+
+_TL;DR_: Griffel relies on the order of CSS classes, which can lead to clashes when multiple bundles are involved. See the simple example below:
```

Comment:
I assume this same limitations is present in any exisitng CSS in JS solution stylex included correct ?

Reply:
StyleX does not have this issue as they rely on `@layer`, examples:
- using `@layer`: https://stackblitz.com/edit/vitejs-vite-yq22zf
- just sorting: https://stackblitz.com/edit/vitejs-vite-xldehe

(difference in behavior is "A+B" frame)

> Note: Stylex also polyfill mode as `@layer`s are not widely supported that also has a proper behavior (https://stackblitz.com/edit/vitejs-vite-jinv4b). However, doing `:not(#\#):not(#\#):not(#\#):not(#\#)` (and more) is a real performance concern.

---

I have evaluated an option to use `@layer` and there are two issues: the browser support & the fact that `@layer` intentionally does not handle specificity between layers, this results in odd visual diffs once any selector is used in a rule. As Stylex has a very limited selector support, it's less a problem for them.

> Note: StyleX has own quirks with sorting of shorthands (border-* behavior).

Reply:
Layer is also problematic when you want to combine solutions that don't use layer. According to [layer specs](https://developer.mozilla.org/en-US/docs/Web/CSS/@layer#description)

> This means that any styles declared outside of a layer will override styles declared in a layer, regardless of specificity.

Once you have global CSS - they are more specific than any layer on the page